### DISCUSSION TITLE
:extensions:common:json-ld:test FAILED

### DISCUSSION THREAD

**Discussion by rohrbachd** [eclipse-edc/Samples#171] [Q&A]
Hi,

I am new to the project. I cloned the EDC connector and was trying to build it using `./gradlew clean build`. I am using java 17 and gradle 8,4 on Windows. The build seems to run successfully until the unit test  `:extensions:common:json-ld:test` . Any help would be appreciated.

The following is the output from the tests that failed:

`> Task :extensions:common:json-ld:test
Java HotSpot(TM) 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended

JsonLdExtensionTest > verifyCachedDocsFromConfig_multipleEntries_oneIncomplete(ServiceExtensionContext, JsonLdExtension) FAILED
    java.lang.AssertionError:
    Expecting map:
      {"http://foo.org/doc.json"=file:/C:/tmp/foo/doc.json, "http://www.w3.org/ns/odrl.jsonld"=file:/C:/Development/EDC/Connector/extensions/common/json-ld/build/resources/main/document%5codrl.jsonld}
    to contain entries:
      ["http://foo.org/doc.json"=file:/tmp/foo/doc.json]
    but the following map entries had different values:
      ["http://foo.org/doc.json"=file:/C:/tmp/foo/doc.json (expected: file:/tmp/foo/doc.json)]
        at org.eclipse.edc.jsonld.JsonLdExtensionTest.verifyCachedDocsFromConfig_multipleEntries_oneIncomplete(JsonLdExtensionTest.java:104)

JsonLdExtensionTest > verifyCachedDocsFromConfig_oneValidEntry_withSuperfluous(ServiceExtensionContext, JsonLdExtension) FAILED
    java.lang.AssertionError:
    Expecting map:
      {"http://foo.org/doc.json"=file:/C:/tmp/foo/doc.json, "http://www.w3.org/ns/odrl.jsonld"=file:/C:/Development/EDC/Connector/extensions/common/json-ld/build/resources/main/document%5codrl.jsonld}
    to contain entries:
      ["http://foo.org/doc.json"=file:/tmp/foo/doc.json]
    but the following map entries had different values:
      ["http://foo.org/doc.json"=file:/C:/tmp/foo/doc.json (expected: file:/tmp/foo/doc.json)]
        at org.eclipse.edc.jsonld.JsonLdExtensionTest.verifyCachedDocsFromConfig_oneValidEntry_withSuperfluous(JsonLdExtensionTest.java:70)

JsonLdExtensionTest > verifyCachedDocsFromConfig_multipleEntries_oneInvalid(ServiceExtensionContext, JsonLdExtension) FAILED
    java.lang.AssertionError:
    Expecting map:
      {"http://foo.org/doc.json"=file:/C:/tmp/foo/doc.json, "http://www.w3.org/ns/odrl.jsonld"=file:/C:/Development/EDC/Connector/extensions/common/json-ld/build/resources/main/document%5codrl.jsonld}
    to contain entries:
      ["http://foo.org/doc.json"=file:/tmp/foo/doc.json]
    but the following map entries had different values:
      ["http://foo.org/doc.json"=file:/C:/tmp/foo/doc.json (expected: file:/tmp/foo/doc.json)]
        at org.eclipse.edc.jsonld.JsonLdExtensionTest.verifyCachedDocsFromConfig_multipleEntries_oneInvalid(JsonLdExtensionTest.java:122)

JsonLdExtensionTest > verifyCachedDocsFromConfig_multipleValidEntries(ServiceExtensionContext, JsonLdExtension) FAILED
    java.lang.AssertionError:
    Expecting map:
      {"http://bar.org/doc.json"=file:/C:/tmp/bar/doc.json, "http://foo.org/doc.json"=file:/C:/tmp/foo/doc.json, "http://www.w3.org/ns/odrl.jsonld"=file:/C:/Development/EDC/Connector/extensions/common/json-ld/build/resources/main/document%5codrl.jsonld}
    to contain entries:
      ["http://foo.org/doc.json"=file:/tmp/foo/doc.json]
    but the following map entries had different values:
      ["http://foo.org/doc.json"=file:/C:/tmp/foo/doc.json (expected: file:/tmp/foo/doc.json)]
        at org.eclipse.edc.jsonld.JsonLdExtensionTest.verifyCachedDocsFromConfig_multipleValidEntries(JsonLdExtensionTest.java:87)

JsonLdExtensionTest > verifyCachedDocsFromConfig_oneValidEntry(ServiceExtensionContext, JsonLdExtension) FAILED
    java.lang.AssertionError:
    Expecting map:
      {"http://foo.org/doc.json"=file:/C:/tmp/foo/doc.json, "http://www.w3.org/ns/odrl.jsonld"=file:/C:/Development/EDC/Connector/extensions/common/json-ld/build/resources/main/document%5codrl.jsonld}
    to contain entries:
      ["http://foo.org/doc.json"=file:/tmp/foo/doc.json]
    but the following map entries had different values:
      ["http://foo.org/doc.json"=file:/C:/tmp/foo/doc.json (expected: file:/tmp/foo/doc.json)]
        at org.eclipse.edc.jsonld.JsonLdExtensionTest.verifyCachedDocsFromConfig_oneValidEntry(JsonLdExtensionTest.java:54)`

**Comment by rohrbachd** [comment]
Other tests are also failing. I was able to run everything successfully with the following

`./gradlew build -x :extensions:common:json-ld:test -x :extensions:common:api:management-api-configuration:test -x :extensions:control-plane:api:management-api:contract-negotiation-api:test -x :extensions:control-plane:api:management-api:policy-definition-api:test`

**Comment by ndr-brt** [comment]
this is not related to the samples, in fact there's an issue on the Connector repository about this: https://github.com/eclipse-edc/Connector/issues/3408

**Reply by rohrbachd** [reply]
Thanks a lot for the link. Much appreciated. I have a look. Perhaps I can fix it.
